src/v/cluster/tm_stm.h

```diff
@@ -139,7 +139,13 @@ class tm_stm final : public persisted_stm {
     // get_end_lock method returns a mutex used to coordinate start
     // and end of a transaction
     ss::lw_shared_ptr<mutex> get_end_lock(kafka::transactional_id tid) {
-        return _end_locks.find(tid)->second;
+        auto lock_it = _end_locks.find(tid);
```

Comment:
I think that in here you can always return iterator returned by `try_emplace`

Reply:
Do you mean something like?

```
auto [lock_it, _] = _end_locks.try_emplace(
  tid, ss::make_lw_shared<mutex>());
return lock_it;
```

The problem with this approach is that in most cases (a lock is already in the hash table) it creates an object `ss::make_lw_shared<mutex>()` and immediately discards it. It may create unnecessary memory pressure so I decided to avoid it.